runhouse/resources/hardware/cluster.py

```diff
@@ -948,24 +898,61 @@ def _rsync(
 
         ssh_credentials = copy.copy(self.ssh_creds())
         ssh_credentials.pop("ssh_host", self.address)
+        pwd = ssh_credentials.pop("password", None)
 
-        if not ssh_credentials.get("password"):
-            # Use SkyPilot command runner
-            if not ssh_credentials.get("ssh_private_key"):
-                ssh_credentials["ssh_private_key"] = None
-            runner = SkySSHRunner(self.address, **ssh_credentials)
+        runner = SkySSHRunner(self.address, **ssh_credentials)
+        if not pwd:
             if up:
                 runner.run(["mkdir", "-p", dest], stream_logs=False)
             else:
                 Path(dest).expanduser().parent.mkdir(parents=True, exist_ok=True)
             runner.rsync(
-                source, dest, up=up, filter_options=filter_options, stream_logs=False
+                source,
+                dest,
+                up=up,
+                filter_options=filter_options,
+                stream_logs=stream_logs,
             )
         else:
-            if dest.startswith("~/"):
-                dest = dest[2:]
+            import pexpect
 
-            self._fsspec_sync(source, dest, up)
+            if up:
+                ssh_command = runner.run(
+                    ["mkdir", "-p", dest],
+                    stream_logs=False,
+                    return_cmd=True,
+                    ssh_mode=SshMode.INTERACTIVE,
+                )
+                ssh = pexpect.spawn(ssh_command, encoding="utf-8")
+                ssh.logfile_read = sys.stdout
+                # If CommandRunner uses the control path, the password may not be requested
+                next_line = ssh.expect(["assword:", pexpect.EOF])
```

Comment:
not sure if typo or meant for some regex match

Reply:
Intended, matches both "Password:" and "password:"